tests/cache/tests.py

```diff
@@ -1833,16 +1833,13 @@ def test_cache_key_no_i18n(self):
     @override_settings(USE_I18N=False, USE_L10N=False, USE_TZ=True)
     def test_cache_key_with_non_ascii_tzname(self):
         # Regression test for #17476
-        class CustomTzName(timezone.UTC):
-            name = ''
-
-            def tzname(self, dt):
-                return self.name
+        class CustomTzName(timezone.utc):
```

Comment:
I tried something else in the follow up commit but it seems to leak the custom attributes to on the original instance (`deepcopy` seems to have the same problem). Anyway, if you don't have another diea, we could add a comment that the code is a bit artificial but simulates the correct conditions for the test.